### DISCUSSION TITLE
Problem with extractors substitution

### DISCUSSION THREAD

**Discussion by Tutayyy** [projectdiscovery/nuclei#4890] [Q&A]
I'm creating custom templates for each Business scenarios in app. Business scenarios may include several steps. Business scenarios may include several steps. To fuzz the request parameters, for example, step 2, I have to send a valid request for step 1 and a request with payload for step 2. In response to the 1st request, I receive a nonce operation identifier (operationID), which I use in request 2. When I launch nuclei, an extra iteration of the loop occurs, i.e. within the first iteration, in response to step 1, I receive operationID[1], step 2 is not sent; within the second iteration I get operationID[2]; in the request for step 2, the identifier of operation 1 is used (and the identifier of operationID[2] is needed); within the 3rd iteration I receive the identifier of operation 3, in the request for step 2 the identifier of operationID[2] is used (and the identifier of operationID[3] is needed). I'm using threads: 1

**Comment by olearycrew** [comment]
I'm not entirely sure if I understand what you need to do, but I recommend looking at both workflows and the JavaScript protocol:

* https://docs.projectdiscovery.io/templates/protocols/javascript/introduction
* https://docs.projectdiscovery.io/templates/workflows/overview

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
